src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ItemMapper.java

```diff
@@ -26,11 +29,13 @@
 class ItemMapper {
 
     private final FieldMapper fieldMapper;
-    private final TableMapping tableMapping;
+    private final Map<String, List<FieldMapping>> virtualToPhysicalFieldMappings;
+    private final Map<String, List<FieldMapping>> physicalToVirtualFieldMappings;
 
-    ItemMapper(TableMapping tableMapping, FieldMapper fieldMapper) {
+    ItemMapper(FieldMapper fieldMapper, Map<String, List<FieldMapping>> virtualToPhysicalFieldMappings) {
```

Comment:
I did some refactoring here, instead of taking a `TableMapping` in the constructor, it takes a field mapping.  This allows me to re-use this class for mapping the item and keys.